Dockerfile

```diff
@@ -36,25 +30,27 @@ RUN mkdir -p $PKG_PATH/src && touch $PKG_PATH/src/$PKG_NAME.jl
 
 # Install and build the package requirements. Record any system packages that need to be
 # installed in order to build any dependencies which is helpful for future maintenence.
-RUN julia -e " \
+# Note: For packages specified in the Project.toml (e.g. AWSBatch and Memento) we can skip
+# specifying a version number here.
+RUN julia -e ' \
     using Pkg; \
     Pkg.update(); \
-    Pkg.develop(PackageSpec(name=\"$PKG_NAME\", path=\"$PKG_PATH\")); \
-    Pkg.add([\"AWSBatch\", \"Memento\"]); \
-    "
+    Pkg.develop(PackageSpec(name=ENV["PKG_NAME"], path=ENV["PKG_PATH"])); \
+    Pkg.add(["AWSBatch", "Memento"]); \
+    '
 
 # Control if pre-compilation is run when new Julia packages are installed.
-ARG PRECOMPILE="true"
+ARG PKG_PRECOMPILE="true"
 
 # Perform precompilation of packages.
-RUN if [[ "$PKG_PRECOMPILE" == "true" ]]; then \
-        $HOME/precompile.sh; \
+RUN if [ "$PKG_PRECOMPILE" = "true" ]; then \
```

Comment:
Is this needed, since we have the same check down on line 78 when creating the system image?

Reply:
This precompile call is here to precompile all of the dependencies of AWSClusterManagers.jl. If someone is experimenting with enabling/disabling `CREATE_SYSIMG` then this earlier step saves time. We could remove the second precompile call but that one is just to compile AWSClusterManagers.jl again as the contents of the package have since been modified